esphome/components/seeed_mr24hpc1/seeed_mr24hpc1.cpp

```diff
@@ -0,0 +1,1030 @@
+    this->custom_mode_end_text_sensor_->publish_state("Set Success!");
+  } else if (data[FRAME_COMMAND_WORD_INDEX] == 0x89) {
+    if (data[FRAME_DATA_INDEX] == 0) {
+      if (this->custom_mode_end_text_sensor_ != nullptr) {
+        this->custom_mode_end_text_sensor_->publish_state("Not in custom mode");
+      }
+      if (this->custom_mode_number_ != nullptr) {
+        this->custom_mode_number_->publish_state(0);
+      }
+      if (this->custom_mode_num_sensor_ != nullptr) {
+        this->custom_mode_num_sensor_->publish_state(data[FRAME_DATA_INDEX]);
+      }
+    }
+  } else {
+    ESP_LOGD(TAG, "[%s] No found COMMAND_WORD(%02X) in Frame", __FUNCTION__, data[FRAME_COMMAND_WORD_INDEX]);
+  }
+}
+
+void MR24HPC1Component::r24_frame_parse_human_information(uint8_t *data) {
+  if ((this->someoneExists_binary_sensor_ != nullptr) &&
+      ((data[FRAME_COMMAND_WORD_INDEX] == 0x01) || (data[FRAME_COMMAND_WORD_INDEX] == 0x81))) {
+    this->someoneExists_binary_sensor_->publish_state(S_SOMEONE_EXISTS_STR[data[FRAME_DATA_INDEX]]);
+  } else if ((this->motion_status_text_sensor_ != nullptr) &&
+             ((data[FRAME_COMMAND_WORD_INDEX] == 0x02) || (data[FRAME_COMMAND_WORD_INDEX] == 0x82))) {
+    if (data[FRAME_DATA_INDEX] < 3) {
+      this->motion_status_text_sensor_->publish_state(S_MOTION_STATUS_STR[data[FRAME_DATA_INDEX]]);
+    }
+  } else if ((this->movementSigns_sensor_ != nullptr) &&
+             ((data[FRAME_COMMAND_WORD_INDEX] == 0x03) || (data[FRAME_COMMAND_WORD_INDEX] == 0x83))) {
+    this->movementSigns_sensor_->publish_state(data[FRAME_DATA_INDEX]);
+  } else if ((this->unman_time_select_ != nullptr) &&
+             ((data[FRAME_COMMAND_WORD_INDEX] == 0x0A) || (data[FRAME_COMMAND_WORD_INDEX] == 0x8A))) {
+    // none:0x00  1s:0x01 30s:0x02 1min:0x03 2min:0x04 5min:0x05 10min:0x06 30min:0x07 1hour:0x08
+    if (data[FRAME_DATA_INDEX] < 9) {
+      this->unman_time_select_->publish_state(S_UNMANNED_TIME_STR[data[FRAME_DATA_INDEX]]);
+    }
+  } else if ((this->keep_away_text_sensor_ != nullptr) &&
+             ((data[FRAME_COMMAND_WORD_INDEX] == 0x0B) || (data[FRAME_COMMAND_WORD_INDEX] == 0x8B))) {
+    // none:0x00  close_to:0x01  far_away:0x02
+    if (data[FRAME_DATA_INDEX] < 3) {
+      this->keep_away_text_sensor_->publish_state(S_KEEP_AWAY_STR[data[FRAME_DATA_INDEX]]);
+    }
+  } else {
+    ESP_LOGD(TAG, "[%s] No found COMMAND_WORD(%02X) in Frame", __FUNCTION__, data[FRAME_COMMAND_WORD_INDEX]);
+  }
+}
+
+// Print data frame
+static void show_frame_data(const uint8_t *data, int len) {
+  printf("[%s] FRAME: %d, ", __FUNCTION__, len);
+  for (int i = 0; i < len; i++) {
+    printf("%02X ", data[i] & 0xff);
+  }
+  printf("\r\n");
+}
+
+// Sending data frames
+void MR24HPC1Component::send_query(uint8_t *query, size_t string_length) {
+  int i;
+  for (i = 0; i < string_length; i++) {
+    write(query[i]);
+  }
+  show_frame_data(query, i);
```

Comment:
Please make sure all member variables and functions are prefixed with `this->` as per our published [codebase standards](https://esphome.io/guides/contributing#codebase-standards)
```suggestion
  this->show_frame_data(query, i);
```